api/scenario_api_simulator/src/scenario_api_simulator.cpp

```diff
@@ -16,28 +16,50 @@
 
 #include <scenario_api_simulator/scenario_api_simulator.h>
 
-ScenarioAPISimulator::ScenarioAPISimulator() : nh_(""), pnh_("~")
+#include <autoware_perception_msgs/msg/semantic.hpp>
+#include <autoware_perception_msgs/msg/shape.hpp>
+#include <dummy_perception_publisher/msg/initial_state.hpp>
+#include <scenario_api_utils/scenario_api_utils.h>
+#include <unique_identifier_msgs/msg/uuid.hpp>
+
+#include <functional>
+#include <random>
+
+ScenarioAPISimulator::ScenarioAPISimulator()
```

Comment:
This should instead take rclcpp::Node::SharedPtr as an input to the constructor and use it for setting parameters and pub/subs.

ScenarioAPISimulator is a member class used by scenario_api_package, which means original ROS1 code registers ScenarioAPI class as one package whereas making ScenarioAPISimulator inherit rclcpp::Node class would make this a separate node.
https://github.com/tier4/scenario_runner.iv.universe/blob/d6ec250b2ff9a0ae879ca19a27a7fc3599af4a0a/api/scenario_api/include/scenario_api/scenario_api_core.h#L305-L307

Similar discussion was made in this PR: 
https://github.com/tier4/scenario_runner.iv.universe/pull/12#discussion_r521889886

Reply:
Sure, no problem.